README.md

```diff
@@ -6,36 +6,149 @@
+                                              <div>Movie stuff...</div>
+                                              <div>Movie stuff...</div></div>} />
+```
 
-Then start walking through the `React.Children` API. These methods should be
-easy to understand once students have a grip on how `this.props.children` works.
+_And_ we'd have to wrap it in an enclosing `div`! Thankfully, we can just nest it inside of the component like we did
+above, much like we nest regular HTML elements.
 
+## React.Children
+Since `this.props.children` can have one element, multiple elements or none at all, its value is respectively
+`undefined`, a single child node, or an array of child nodes. Sometimes, we want to transform our children before
+rendering them — for example, to add additional props to every child. If we wanted to do that, we'd have to take into
+account what the possible types of `this.props.children` are. For example, if there is only one child, we can't map it.
 
-## Resources
+Luckily, React provides us with a clean API to handle of looping children. If there is only one child (or none at all),
+it won't throw a fuss — it'll handle things for us nicely in the background.
+
+Let's say we have a list of `Movie` components that are nested inside of a `MovieBrowser` component:
+
+```js
+<MovieBrowser>
+  <Movie title="Mad Max: Fury Road" />
+  <Movie title="Harry Potter & The Goblet Of Fire" />
+</MovieBrowser>
+```
+
+Now, let's assume for some reason that we need to pass down an extra prop to our children — the props would like to know
+if they are being played or not. Our `MovieBrowser` component would look something like this, before we added the prop:
+
+```js
+export default class MovieBrowser extends React.Component {
+  render() {
+    const currentPlayingTitle = 'Mad Max: Fury Road';
+    
+    return (
+      <div className="movie-browser">
+        {this.props.children}
+      </div>      
+    );
+  }
+}
+```
+
+Now let's add in our `isPlaying` prop to the children of `MovieBrowser`:
+
+```js
+export default class MovieBrowser extends React.Component {
+  render() {
+    const currentPlayingTitle = 'Mad Max: Fury Road';
+    const childrenWithExtraProp = React.Children.map(this.props.children, child => {
+      return React.cloneElement(child, {
+        isPlaying: child.props.title === currentPlayingTitle
+      });
+    });
+    
+    return (
+      <div className="movie-browser">
+        {childrenWithExtraProp}
+      </div>      
+    );
+  }
+}
+```
 
-- [React.Children](https://facebook.github.io/react/docs/top-level-api.html#react.children)
+`React.Children.map` has two parameters: the first one is the children themselves, and the second one is a function that
+transforms the value of the child. In this case, we're adding an extra prop. We do that using `React.cloneElement`. As
+the first argument we pass in the component itself (the child), and as the second argument, we pass in any additional
```

Comment:
"we pass in the component itself (the child)" -> "we pass in the child component itself" ?